packages/plugin-link/src/Link.ts

```diff
@@ -73,32 +73,54 @@ export class Link<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T>
 
     async link(params: LinkParams): Promise<void> {
         // If the url is undefined, ask the user to provide one.
+        const range = params.context.range;
+        const selectedInlines = range.selectedNodes(InlineNode);
+        const label = params.label || selectedInlines.map(node => node.textContent).join('');
+
         if (!params.url) {
+            const firstLink = selectedInlines.find(node => node.modifiers.find(LinkFormat));
+            const link = firstLink && firstLink.modifiers.find(LinkFormat).clone();
+            const url = link?.url || '';
+
             const layout = this.editor.plugins.get(Layout);
             await layout.remove('link');
-            await layout.add('link');
-
-            return this.editor.execCommand('show', { componentID: 'link' });
+            return layout.add('link', undefined, { label: label, url: url });
         }
 
         // Otherwise create a link and insert it.
         const link = new LinkFormat(params.url);
         return this.editor.execCommand<Char>('insertText', {
-            text: params.label || link.url,
+            text: label || link.url,
             formats: new Modifiers(link),
         });
     }
     unlink(params: LinkParams): void {
         const range = params.context.range;
-        const node = range.start.previousSibling() || range.start.nextSibling();
-        if (!node.is(InlineNode)) return;
 
-        const link = node.modifiers.find(LinkFormat);
-        if (!link) return;
+        if (range.isCollapsed()) {
+            const previousNode = range.start.previousSibling();
+            const nextNode = range.start.nextSibling();
+            const node = Link.isLink(previousNode) ? previousNode : nextNode;
+            if (!Link.isLink(node)) return;
+
+            const link = node.modifiers.find(LinkFormat);
+            const sameLink: Typeguard<InlineNode> = Link.isLink.bind(Link, link);
+            this._removeLinkOnNodes([node, ...node.adjacents(sameLink)]);
+        } else {
+            const selectedNodes = range.selectedNodes(InlineNode);
+            // remove the format 'LinkFormat' for all nodes.
+            this._removeLinkOnNodes(selectedNodes);
+        }
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
 
-        const sameLink: Typeguard<InlineNode> = Link.isLink.bind(Link, link);
-        for (const inline of [node, ...node.adjacents(sameLink)]) {
-            inline.modifiers.remove(link);
+    _removeLinkOnNodes(nodes: VNode[]): void {
```

Comment:
doc please :-)

Reply:
This comment should not have been resolved as it is still relevant as of now. I'm unresolving it.

Reply:
The comment below is not clear enough ?

Reply:
Only proper function doc written above the function in `/**` format is read by automated documentation systems like `tsdoc`. This is why we (try to) document all our functions in that way. You can open other files to see examples of it. Please note that we do not document the types of the parameters since this is redundant with the typing of Typescript which is correctly parsed by `tsdoc`.